src/models/view/sendAccessView.ts

```diff
@@ -0,0 +1,24 @@
+import { SendType } from '../../enums/sendType';
+
+import { SendAccess } from '../domain/sendAccess';
+
+import { SendFileView } from './sendFileView';
+import { SendTextView } from './sendTextView';
+import { View } from './view';
+
+export class SendAccessView implements View {
+    id: string = null;
+    name: string = null;
+    type: SendType = null;
+    text = new SendTextView();
+    file = new SendFileView();
+
+    constructor(s?: SendAccess) {
+        if (!s) {
+            return;
+        }
+
+        this.id = s.id;
+        this.type = s.type;
```

Comment:
Wouldn't you also want to set `this.name = s.name`?

Reply:
No, the pattern I followed has all non-encrypted things being set in the ctor like this while the encrypted values get decrypted and assigned in another function.